src/tools/tests/bundle-test.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import fs from 'fs';
+import JSZip from 'jszip';
+
+import {bundle, bundleListing} from '../bundle.js';
+import {assert} from '../../platform/chai-web.js';
+
+describe('Bundle Tool', () => {
+  it('bundles particle source files', async () => {
+    assert.deepEqual(await bundleListing('src/tools/tests/test-data/a/a.manifest'), [
+      {filePath: 'src/tools/tests/test-data/a/a.js', bundlePath: 'a.js', entryPoint: false},
+      {filePath: 'src/tools/tests/test-data/a/a.manifest', bundlePath: 'a.manifest', entryPoint: true},
+    ]);
+  });
+  it('bundles store json files', async () => {
+    assert.deepEqual(await bundleListing('src/tools/tests/test-data/b/b.manifest'), [
+      {filePath: 'src/tools/tests/test-data/b/b.json', bundlePath: 'b.json', entryPoint: false},
+      {filePath: 'src/tools/tests/test-data/b/b.manifest', bundlePath: 'b.manifest', entryPoint: true},
+    ]);
+  });
+  it('handles manifest imports', async () => {
+    assert.deepEqual(await bundleListing('src/tools/tests/test-data/c/c.manifest'), [
+      {filePath: 'src/tools/tests/test-data/a/a.js', bundlePath: 'a/a.js', entryPoint: false},
+      {filePath: 'src/tools/tests/test-data/a/a.manifest', bundlePath: 'a/a.manifest', entryPoint: false},
+      {filePath: 'src/tools/tests/test-data/b/b.json', bundlePath: 'b/b.json', entryPoint: false},
+      {filePath: 'src/tools/tests/test-data/b/b.manifest', bundlePath: 'b/b.manifest', entryPoint: false},
+      {filePath: 'src/tools/tests/test-data/c/c.manifest', bundlePath: 'c/c.manifest', entryPoint: true},
+    ]);
+  });
+  it('accepts multiple entry points', async () => {
+    assert.deepEqual(
+        await bundleListing(
+            'src/tools/tests/test-data/a/a.manifest',
+            'src/tools/tests/test-data/b/b.manifest'),
+        [
+          {filePath: 'src/tools/tests/test-data/a/a.js', bundlePath: 'a/a.js', entryPoint: false},
+          {filePath: 'src/tools/tests/test-data/a/a.manifest', bundlePath: 'a/a.manifest', entryPoint: true},
+          {filePath: 'src/tools/tests/test-data/b/b.json', bundlePath: 'b/b.json', entryPoint: false},
+          {filePath: 'src/tools/tests/test-data/b/b.manifest', bundlePath: 'b/b.manifest', entryPoint: true},
+        ]
+    );
+  });
+  it('bundles Products demo', done => {
+    bundle(['src/runtime/test/artifacts/Products/Products.recipes'], 'test-output/bundle/products.zip', false).then(() => {
+      fs.readFile('test-output/bundle/products.zip', (err, data) => {
```

Comment:
fs.readFileSync?

Reply:
Right! I managed to simplify the async stuff here. I don't remember know why I thought this has to be so complex :/